src/pages/JoinMembership/JoinMembership.tsx

```diff
@@ -30,13 +30,21 @@ const JoinMembershipPage = () => {
         email: form.email,
         password: form.password
       });
-      let accessToken: string | undefined = undefined;
-      if ('data' in loginRes && loginRes.data && 'accessToken' in loginRes.data) {
-        accessToken = (loginRes.data as { accessToken?: string }).accessToken;
-      }
-      if (accessToken) {
-        login(accessToken);
+
+      // 로그인 응답에서 필요한 데이터 추출
+      if ('data' in loginRes && loginRes.data) {
+        const { accessToken, name, userId, roleCode } = loginRes.data as {
+          accessToken?: string;
+          name?: string;
+          userId?: number;
+          roleCode?: number;
+        };
+
+        if (accessToken && name && userId && roleCode) {
```

Comment:
```suggestion
        if (accessToken && name && userId != null && roleCode != null) {
```
일반 유저의 roleCode가 0이라서 false가 아님에도 false를 반환해서 문제가 생기는거 같습니다.